operator/pkg/ciliumidentity/controller_test.go

```diff
@@ -197,3 +199,226 @@ func verifyCIDUsageInCES(ctx context.Context, fakeClient *k8sClient.FakeClientse
+				{cid1.SecurityLabels},
+				// TODO: Update the test to actually delete CID after the pod is deleted.
+				// https://github.com/cilium/cilium/issues/34740
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "recreate_cid_on_pod_update",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"update", updatedPod, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				// TODO: Update the test to actually delete CID after the pod is deleted.
+				// https://github.com/cilium/cilium/issues/34740
+				{cid1.SecurityLabels, cid2.SecurityLabels},
+			},
+		},
+		{
+			name: "reuse_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"delete", pod1, nil},
+				{"create", pod1, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "update_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{},
+			},
+			cidUpdateList: []*capi_v2.CiliumIdentity{
+				nil,
+				cid2,
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid2.SecurityLabels},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cid, _, fakeClient, _, h := initHiveTest(true)
+			ctx := context.Background()
+			tlog := hivetest.Logger(t)
+			if err := h.Start(tlog, ctx); err != nil {
+				t.Fatalf("starting hive encountered an error: %s", err)
+			}
+			defer func() {
+				if err := h.Stop(tlog, ctx); err != nil {
+					t.Fatalf("stopping hive encountered an error: %v", err)
+				}
+			}()
+
+			if _, err := fakeClient.Slim().CoreV1().Namespaces().Create(ctx, ns1, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("create namespace: %v", err)
+			}
+
+			cidStore, _ := (*cid).Store(ctx)
+
+			// Test starts here.
+			// Run podActions in the sequence defined in the test case.
+			// Verify CID state after each podAction.
+			for actionIdx, podAction := range tc.podActions {
+				pod := podAction.podObj
+				switch podAction.action {
+				case "create":
+					if _, err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+						t.Fatalf("create pod: %v", err)
+					}
+				case "update":
+					if _, err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Update(ctx, pod, metav1.UpdateOptions{}); err != nil {
+						t.Fatalf("update pod: %v", err)
+					}
+				case "delete":
+					if err := fakeClient.Slim().CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
+						t.Fatalf("delete pod: %v", err)
+					}
+				}
+
+				if len(tc.cidUpdateList) > 0 && tc.cidUpdateList[actionIdx] != nil {
+					cidObjList := cidStore.List()
+					if len(cidObjList) == 0 {
+						t.Fatalf("listed CIDs = 0, want >= 1")
+					}
+
+					cidToUpdate := cidObjList[0]
```

Comment:
The `cidObjList[0]` could be a random CID, should we attempt to get the `cidToUpdate` by an unique key?

Reply:
Done.